framework/src/relationshipmanagers/AugmentSparsityOnInterface.C

```diff
@@ -227,21 +227,70 @@ AugmentSparsityOnInterface::operator()(const MeshBase::const_element_iterator &
     {
       ghost_mortar_interface_couplings(elem);
 
-      if (_ghost_point_neighbors && (elem->subdomain_id() == secondary_subdomain_id))
```

Comment:
Can't understand why the second condition is removed.

Reply:
It's related to this comment:
```c++
// I hypothesize that node processor ids are tied to higher dimensional element processor
// ids over lower dimensional element processor ids based on debugging experience.
// Consequently we need to be checking for higher dimensional elements and whether they are
// along our secondary boundary. From there we will query the AMG object's
// mortar-interface-coupling container to get the secondary lower-dimensional element, and
// then we will ghost it's point neighbors and their mortar interface couples
```
Essentially we are going to initiate our search for point neighbors starting with higher dimensional elements along the secondary boundary, *not* from lower dimensional elements along the secondary subdomain

Reply:
Node processor ids are what are important for our mortar nodal aux kernel evaluations